app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/OfflineRerouteActivity.java

```diff
@@ -282,22 +287,39 @@ private void startNavigation(DirectionsRoute route) {
   }
 
   private void findRoute() {
-    String tilesDirPath = obtainOfflineDirectoryFor("tiles");
-    Timber.d("Tiles directory path: %s", tilesDirPath);
+    getVersionsAndRoute();
+  }
+
+  private void getVersionsAndRoute() {
+    OfflineTileVersions versions = new OfflineTileVersions(Mapbox.getAccessToken());
+
+    versions.getRouteTileVersions(new Callback<RouteTileVersionsResponse>() {
+      @Override
+      public void onResponse(Call<RouteTileVersionsResponse> call, Response<RouteTileVersionsResponse> response) {
+        String version = response.body().availableVersions().get(0);
+        String tilesDirPath = obtainOfflineDirectoryFor("tiles", version);
+        Timber.d("Tiles directory path: %s", tilesDirPath);
+
+        navigation.initializeOfflineData(tilesDirPath, () -> {
+          OfflineRoute offlineRoute = obtainOfflineRoute(origin, destination);
+          route = navigation.findOfflineRoute(offlineRoute);
+          handleNewRoute(route);
+        });
+      }
 
-    offlineRouting.initializeOfflineData(tilesDirPath, () -> {
-      OfflineRoute offlineRoute = obtainOfflineRoute(origin, destination);
-      route = offlineRouting.findOfflineRoute(offlineRoute);
-      handleNewRoute(route);
+      @Override
+      public void onFailure(Call call, Throwable throwable) {
```

Comment:
Same thing, can we do something simple that doesn't result in a dead end?

Reply:
@akitchen It's a lot harder to have a simple solution for this class, but I did add an error toast